chall06/mzomeno-.c

```diff
@@ -0,0 +1,173 @@
+static int get_line_length(char *board)
+{
+  int line_length;
+
+  line_length = 0;
+  while (*board != '\n' && *board != '\0')
+    {
+      if (*board != '.' && *board != 'K' &&
+	  *board != 'P' && *board != 'R' &&
+	  *board != 'B' && *board != 'Q')
+	return (0);
+      line_length++;
+      board++;
+    }
+  return (line_length);
+}
+
+static int check_board(char *board, int line_length)
+{
+  int line_number;
+
+  if (line_length == 0 || line_length > 9)
+    return (0);
+  line_number = 1;
+  while (*board != '\0')
+    {
+      if (line_length != get_line_length(board + (line_length * (line_number + 1))))
+	return (1);
+      line_number++;
+      board++;
+    }
+  if (line_number != line_length)
+    return (0);
+  return (line_number);
+}
+
+static int check_king(char *board)
+{
+  int iterator;
+  int king_position;
+
+  king_position = -1;
+  iterator = 0;
+  while (board[iterator] != '\0')
+    {
+      if (board[iterator] == 'K')
+	{
+	  if (king_position == -1)
+	    king_position = iterator;
+	  else
+	    return (-1);
+	}
+      iterator++;
+    }
+  return (king_position);
+  /*
+   * Note that the function will return -1 if there is no king,
+   * as the king_position variable’s initial value remains untouched
+   */
+}
+
+static int pawn_capture(int line_length, int position, int king_position)
+{
+  if (king_position == (position - line_length - 2) ||
```

Comment:
The Pawn can only capture the King from "the bottom" (check the capture patterns again). Which means `board = "P.\n.K"` should return `1` but and `board = "K.\n.P" should return `0`.